webapp/src/components/backstage/playbook_editor/controls.tsx

```diff
@@ -258,33 +257,24 @@ const LEARN_PLAYBOOKS_TITLE = 'Learn how to use playbooks';
 export const playbookIsTutorialPlaybook = (playbookTitle?: string) => playbookTitle === LEARN_PLAYBOOKS_TITLE;
 
 export const RunPlaybook = ({playbook}: ControlProps) => {
+    const dispatch = useDispatch();
     const {formatMessage} = useIntl();
     const team = useSelector<GlobalState, Team>((state) => getTeam(state, playbook?.team_id || ''));
-    const currentUser = useSelector(getCurrentUser);
     const isTutorialPlaybook = playbookIsTutorialPlaybook(playbook.title);
     const hasPermissionToRunPlaybook = useHasPlaybookPermission(PlaybookPermissionGeneral.RunCreate, playbook);
     const enableRunPlaybook = playbook.delete_at === 0 && hasPermissionToRunPlaybook;
 
-    const runPlaybook = async () => {
-        if (playbook && isTutorialPlaybook) {
-            const playbookRun = await createPlaybookRun(playbook.id, currentUser.id, playbook.team_id, `${currentUser.username}'s onboarding run`, playbook.description);
-            const channel = await Client4.getChannel(playbookRun.channel_id);
-
-            navigateToUrl({
-                pathname: `/${team.name}/channels/${channel.name}`,
-                search: '?forceRHSOpen&openTakeATourDialog',
-            });
-            return;
-        }
-        if (playbook?.id) {
-            telemetryEventForPlaybook(playbook.id, 'playbook_dashboard_run_clicked');
-            navigateToUrl(`/${team.name || ''}/_playbooks/${playbook?.id || ''}/run`);
-        }
-    };
-
     return (
         <PrimaryButtonLarger
-            onClick={runPlaybook}
+            onClick={() => {
+                dispatch(openPlaybookRunModal(
+                    playbook.id,
+                    playbook.default_owner_id,
```

Comment:
Need to account for the action settting bit.

```suggestion
                    playbook.default_owner_enabled ? playbook.default_owner_id : null,
```

Reply:
Alternatively you could pass both values through and do the check in the modal component instead of passing `null`. 0/5 though, @trilopin which way do you think is cleaner?